lib/rubygems/remote_fetcher.rb

```diff
@@ -107,6 +107,9 @@ def api_endpoint(uri)
     rescue Resolv::ResolvError => e
       verbose "Getting SRV record failed: #{e}"
       uri
+    rescue SocketError => e
```

Comment:
Why not add this to the rescue a few lines up so we don't have to duplicate code?